main.c

```diff
@@ -48,20 +47,61 @@ char type_to_char(enum tile_type type);
 int main(void) {
 
     printf("Welcome to Frogger Game!\n");
+
     struct board_tile game_board[SIZE][SIZE];
 
-    // TODO (Stage 1.1) Initialise the gameboard.
-    
-    // Read user input and place turtles.
+    // Initialize the game board
+    initialize_board(game_board);
+
+    // Read user input and place turtles
+    int num_turtles;
     printf("How many turtles? ");
-    // TODO (Stage 1.2): Scan in the turtles, and place them on the map.
+    scanf("%d", &num_turtles);
+    place_turtles(game_board, num_turtles);
 
-    // Start the game and print out the gameboard.
+    // Start the game and print out the gameboard
     printf("Game Started\n");
     print_board(game_board);
 
-    printf("Enter command: ");
-    // TODO (Stage 1.3): Create a command loop, to read and execute commands!
+    // Enter command loop
+    char command;
+    int row, start_col, end_col;
+
+    while (1) {
+        printf("Enter command: ");
+        int result = scanf(" %c", &command);
+        
+        // Exit if CTRL+C or input is invalid
+        if (result == EOF || command == 'q') break;
+        
+        // Handle log command
+        if (command == 'l') {
+            scanf("%d %d %d", &row, &start_col, &end_col);
```

Comment:
<img width="193" alt="image" src="https://github.com/user-attachments/assets/aff189c5-ab7a-48c4-bd76-d2d26f8c1185">

from the requirement the command and the position are entered in one go, probably this will be a bit tricky if we want to extend to different commands, probably need `args1`, `args2`, `args3`, etc.

but this way is okay as well

```suggestion
        int result = scanf("%c %d %d %d", &command, &args1, &args2, &args3);
        
        // Exit if CTRL+C or input is invalid
        if (result == EOF || command == 'q') break;
        
        // Handle log command
        if (command == 'l') {
            scanf("%d %d %d", &row, &start_col, &end_col);
```